### PR TITLE
[feat]: [CDS-82668]: Surfacing GitX errors in case of Remote Flows.

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Changes : Have caught and thrown GitX errors so that sam can propagate to UI.

Testing: GitX errors were observed on Local.

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54636)

### CODE REVIEW

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java [hunk 1]
```diff
@@ -201,6 +204,9 @@ public Environment create(@NotNull @Valid Environment environment) {
       publishEvent(environment.getAccountId(), environment.getOrgIdentifier(), environment.getProjectIdentifier(),
           environment.getIdentifier(), EventsFrameworkMetadataConstants.CREATE_ACTION);
       return createdEnvironment;
+    } catch (ExplanationException | HintException | ScmException ex) {
+      log.error(String.format("Error while saving environment : [%s]", environment.getIdentifier()), ex);
+      throw ex;
     } catch (Exception ex) {
       log.error(String.format("Error while saving environment [%s]", environment.getIdentifier()), ex);
```
Comment (mohitg0795) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java:211]: We should add a different error message here. Right now both error messages are identical and its difficult to interpret difference b/w 2.
eg: "Unexpected error while saving..."

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java [hunk 2]
```diff
@@ -229,6 +232,9 @@ public ServiceEntity create(@NotNull @Valid ServiceEntity serviceEntity) {
       publishEvent(serviceEntity.getAccountId(), serviceEntity.getOrgIdentifier(), serviceEntity.getProjectIdentifier(),
           serviceEntity.getIdentifier(), EventsFrameworkMetadataConstants.CREATE_ACTION);
       return createdService;
+    } catch (ExplanationException | HintException | ScmException ex) {
+      log.error(String.format("Error while saving service: [%s]", serviceEntity.getIdentifier()), ex);
+      throw ex;
     } catch (Exception ex) {
       log.error(String.format("Error while saving service [%s]", serviceEntity.getIdentifier()), ex);
```
Comment (mohitg0795) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java:239]: same as before

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54636)* status: 4 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java` line 211 at r1](https://reviewable.io/reviews/harness/harness-core/54636#gh-1372100363:-NheVGPr3WyQ9buaLn95:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/fc6a0127d9a579c2d1c11a9bd2ca068b7bad678d/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/environment/services/impl/EnvironmentServiceImpl.java#L211)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

We should add a different error message here. Right now both error messages are identical and its difficult to interpret difference b/w 2.
eg: "Unexpected error while saving..."
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java` line 239 at r1](https://reviewable.io/reviews/harness/harness-core/54636#gh-1372100948:-NheWJFS4kyE9XuAibFU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/fc6a0127d9a579c2d1c11a9bd2ca068b7bad678d/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/service/services/impl/ServiceEntityServiceImpl.java#L239)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

same as before
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 1 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54636)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger uts, messagecheck

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 1 of 4 files at r1, 1 of 3 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54636)* status: :shipit: complete! ✓

**Comment by yogesh-chauhan** [comment]
trigger review

**Pull request merged**